design/use_cases/indoor_localization_use_case.md

```diff
@@ -0,0 +1,15 @@
+# Indoor Localization
+As a Robot user I want my robot to know its location on a given map of an indoor area so that it can move around the area
+
+## More details
+- Why is this needed?
+   - This is needed for indoor robot navigation in most (all?) cases
+   - Example: a courier robot in a logistics warehouse
+
+- What is the expected user interaction?
+   - The user should be able to specify a map to use and a location on that map for the robot
+   - The robot should be able to deduce it's own position on a map autonomously
```

Comment:
Well ideally yes it would solve the kidnapped robot problem, but in the cases it doesn't know where it is (which may be often), we want to be able to tell the robot "you are here".

Reply:
There might be a need to physically (programmatically) distinguish from 'location tracker' (amcl, EKF...) and (re)localizer ([DboW](https://github.com/dorian3d/DBow), WIFI? and such), so that they can be distinguished in the overall navigation scheme too.